tracer/src/Datadog.Trace/ClrProfiler/AutoInstrumentation/RabbitMQ/RabbitMQIntegration.cs

```diff
@@ -88,22 +89,23 @@ internal static string GetOperationName(Tracer tracer, string spanKind)
             };
         }
 
-        internal static void SetDataStreamsCheckpointOnProduce(Tracer tracer, Span span, RabbitMQTags tags, IDictionary<string, object> headers)
+        internal static void SetDataStreamsCheckpointOnProduce(Tracer tracer, Span span, RabbitMQTags tags, IBody body, IBasicProperties basicProperties)
```

Comment:
Needs generics again here, though I note the method that calls this isn't necessarily using generics, so feel free to update that one too! 😉 

Alternatively, just pass in the values you need directly, AFAICT all you need is the headers (which was the same as before) and `body.Length`? 

```suggestion
        internal static void SetDataStreamsCheckpointOnProduce<TBody, TBasicProperties>(Tracer tracer, Span span, RabbitMQTags tags, TBody body, TBasicProperties basicProperties)
            where TBody : IBody
            where TBasicProperties : IBasicProperties
```